src/ILCompiler.Compiler/src/Compiler/SingleFileCompilationModuleGroup.cs

```diff
@@ -27,7 +27,11 @@ public override void AddCompilationRoots(IRootingServiceProvider rootProvider)
         {
             base.AddCompilationRoots(rootProvider);
 
-            AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule, rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
```

Comment:
(I actually already experimented with the `ModuleDesc` extensions for this some time ago: https://github.com/MichalStrehovsky/corert/commit/52702ccf365fde15b3f2b6d58df1ea17f99edfc5. I ended up not using this because it wasn't saving anything because the NameMangler loads a bunch of junk types and methods from each assembly anyway.)